### PR TITLE
get rid of use of refs and use onKeyDown

### PR DESCRIPTION
**Pull request by benjaminEwhite**
topic picker was failing in some apps with error
discussed here: https://gist.github.com/jimfb/4faa6cbfb1ef476bd105

after numerous dead ends, turned out that the refs inside
TopicPicker were causing this issue. Easily addressed by getting
rid of refs.

### CODE REVIEW

File: src/TopicPicker/TopicPicker.jsx [hunk 1]
```diff
@@ -224,6 +221,7 @@ class TopicPicker extends React.Component {
 
   render() {
     const {pattern, topics, selectedSuggestionIndex, isFocused} = this.state;
+    const {className, placeholderText} = this.props;
```
Comment (learykara) [inline src/TopicPicker/TopicPicker.jsx]: `className` unused (replace `this.props.className` on line 229)

File: src/TopicPicker/TopicPicker.jsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ class TopicPicker extends React.Component {
     this.state = {
       pattern: '',
       topics: [],
+      topicFormInputValue: null,
```
Comment (learykara) [inline src/TopicPicker/TopicPicker.jsx]: unused

**Comment by learykara** [comment]
lgtm :+1:

File: src/TopicPicker/TopicPicker.jsx (hunk 1 above)
Comment (benjaminEwhite) [inline src/TopicPicker/TopicPicker.jsx]: woops!

File: src/TopicPicker/TopicPicker.jsx (hunk 2 above)
Comment (benjaminEwhite) [inline src/TopicPicker/TopicPicker.jsx]: remnant of earlier attempt to solve this bug. thanks.

**Pull request merged**